readium/shared/src/main/java/org/readium/r2/shared/fetcher/Resource.kt

```diff
@@ -102,6 +104,15 @@ interface Resource : SuspendingCloseable {
     suspend fun readAsXml(): ResourceTry<ElementNode> =
         read().mapCatching { XmlParser().parse(ByteArrayInputStream(it)) }
 
+    /**
+     * Reads the full content as a [Bitmap].
+     */
+    suspend fun readAsBitmap(): ResourceTry<Bitmap> =
```

Comment:
Not necessary for this PR, but this seems to be a useful helper. I used it in the example of the `content.md` user guide to build an index of images.